### DISCUSSION TITLE
Throttle `watch` on contract reads

### DISCUSSION THREAD

**Discussion by mc-ducky** [wevm/wagmi#1252] [Q&A]
I've been using `watch` a lot on the contract reads in the dapp I've recently built. Only after deploying I realised I'm spamming the RPC endpoint really hard. Are there ways to throttle the `watch` functionality, so instead of checking every block, check every time interval, or every other block?

**Comment by tmm** [comment]
What RPC provider are you using? If it doesn't have an associated websocket provider set up, you can set a [`pollingInterval`](https://wagmi.sh/docs/providers/configuring-chains#pollinginterval-optional).
